response.go

```diff
@@ -88,55 +89,41 @@ func (res *Response) Encoding() (encoding.Encoding, error) {
 
 // DetectContentType detects and sets the response content type and charset.
 func (res *Response) DetectContentType(ct string) {
-	// Sensible defaults
-	res.MediaType = "text/plain"
-	res.Charset = ""
-
 	// Autodetect if not passed a Content-Type header
 	if ct == "" {
 		ct = http.DetectContentType(res.Body)
 	}
 
 	// Content type (e.g. text/plain or text/html)
-	mt, params, err := mime.ParseMediaType(ct)
+	mt, _, err := mime.ParseMediaType(ct)
 	if err != nil {
 		return
 	}
 	res.MediaType = mt
-
-	// Character set (e.g. utf-8)
-	cs, ok := params["charset"]
-	if ok {
-		res.Charset = cs
-	}
 	res.DetectCharset()
 }
 
-// DetectCharset returns best guess for the reesponse body character set.
+// DetectCharset sets the charset field of the response to the best guess for
+// the response body's character set.
 func (res *Response) DetectCharset() {
-	// Detect via BOM / HTML meta tag
-	_, cs1, ok1 := charset.DetermineEncoding(res.Body, res.MediaType)
+	var certain bool
 
-	// Detect via ICU
-	cs2, ok2, html := "", false, false
+	// Detect via BOM or HTML media type.
+	_, res.Charset, certain = charset.DetermineEncoding(res.Body, res.MediaType)
+	if certain {
+		return
+	}
+
+	// Detect via ICU and set if confidence is high enough.
 	var det *chardet.Detector
-	if strings.Contains(res.MediaType, "html") || true {
+	if strings.Contains(res.MediaType, "html") {
```

Comment:
We don't get XML anywhere?  Is that perhaps why the original was hard-coding `|| true`?